tests/test_record_aggregator.py

```diff
@@ -0,0 +1,139 @@
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import unittest
+
+from model_analyzer.record.record_aggregator import RecordAggregator
+from model_analyzer.analyzer.perf_analyzer.perf_throughput import PerfThroughput
+from model_analyzer.analyzer.perf_analyzer.perf_latency import PerfLatency
+from model_analyzer.analyzer.perf_analyzer.perf_config import PerfAnalyzerConfig
+
+
+class TestRecordAggregatorMethods(unittest.TestCase):
+    def test_insert_record(self):
+        record_aggregator = RecordAggregator()
+
+        self.assertEqual(record_aggregator.total(), 0)
+
+        throughput_record = PerfThroughput(5, PerfAnalyzerConfig())
+        record_aggregator.insert_record(throughput_record)
+
+        # Assert record is added
+        self.assertEqual(record_aggregator.total(), 1)
+
+    def test_headers(self):
+        record_aggregator = RecordAggregator()
+
+        throughput_record = PerfThroughput(5, PerfAnalyzerConfig())
+        record_aggregator.insert_record(throughput_record)
+
+        self.assertEqual(record_aggregator.headers()[0],
+                         throughput_record.header())
+
+    def test_get_records(self):
+        record_aggregator = RecordAggregator()
+
+        # insert throughput record and check its presence
+        throughput_record = PerfThroughput(5, PerfAnalyzerConfig())
+        record_aggregator.insert_record(throughput_record)
+
+        # Get the record
+        retrieved_records = record_aggregator.get_records()
+        retrieved_throughput = retrieved_records[throughput_record.header()][0]
+
+        self.assertEqual(retrieved_throughput.header(),
+                         throughput_record.header(),
+                         msg="Headers do not match after get_records")
+
+        self.assertEqual(retrieved_throughput.value(),
+                         throughput_record.value(),
+                         msg="Values do not match after get_records")
+
+        # Test for malformed inputs
+        with self.assertRaises(Exception):
+            record_aggregator.get_records(filters=[(lambda x: False)])
+        with self.assertRaises(Exception):
+            record_aggregator.get_records(headers=["header1", "header2"],
+                                          filters=[(lambda x: False)])
+
+        # Insert 2 more throughputs
+        record_aggregator.insert_record(PerfThroughput(1,
+                                                       PerfAnalyzerConfig()))
+        record_aggregator.insert_record(
+            PerfThroughput(10, PerfAnalyzerConfig()))
+
+        # Test get with filters
+        retrieved_records = record_aggregator.get_records(
```

Comment:
Test cases should be specific. You should have a separate one for `get_records_default` and `get_records_filtered`.